system-addon/lib/SectionsManager.jsm

```diff
@@ -77,18 +85,34 @@ const SectionsManager = {
     this.updateSection(id, {enabled: false, rows: []}, true);
     this.emit(this.DISABLE_SECTION, id);
   },
+  updateSections() {
+    this.sections.forEach((section, id) => this.updateSection(id, section, true));
+  },
   updateSection(id, options, shouldBroadcast) {
+    this.updateSectionContextMenuOptions(options);
+
     if (this.sections.has(id)) {
       this.sections.set(id, Object.assign(this.sections.get(id), options));
       this.emit(this.UPDATE_SECTION, id, options, shouldBroadcast);
```

Comment:
Do we care if a section might get an "UPDATE" even if it didn't care about the pref for a context menu item?

Reply:
Sounds like a good improvement once we have a second menu item that's affected by this or a second section that doesn't have SaveToPocket (Highlights will also have SaveToPocket).